aws/resource_aws_chime_voice_connector_group.go

```diff
@@ -0,0 +1,177 @@
+			},
+		},
+	}
+}
+
+func resourceAwsChimeVoiceConnectorGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).chimeconn
+
+	input := &chime.CreateVoiceConnectorGroupInput{
+		Name: aws.String(d.Get("name").(string)),
+	}
+
+	if v, ok := d.GetOk("connector"); ok && len(v.([]interface{})) > 0 {
+		input.VoiceConnectorItems = expandVoiceConnectorItems(v.([]interface{}))
+	}
+
+	resp, err := conn.CreateVoiceConnectorGroupWithContext(ctx, input)
+	if err != nil || resp.VoiceConnectorGroup == nil {
+		return diag.Errorf("error creating Chime Voice connector group: %s", err)
+	}
+
+	d.SetId(aws.StringValue(resp.VoiceConnectorGroup.VoiceConnectorGroupId))
+
+	return resourceAwsChimeVoiceConnectorGroupRead(ctx, d, meta)
+}
+
+func resourceAwsChimeVoiceConnectorGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).chimeconn
+
+	getInput := &chime.GetVoiceConnectorGroupInput{
+		VoiceConnectorGroupId: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetVoiceConnectorGroupWithContext(ctx, getInput)
+	if !d.IsNewResource() && isAWSErr(err, chime.ErrCodeNotFoundException, "") {
+		log.Printf("[WARN] Chime Voice conector group %s not found", d.Id())
+		d.SetId("")
+		return nil
+	}
+	if err != nil || resp.VoiceConnectorGroup == nil {
+		return diag.Errorf("error getting Voice connector group (%s): %s", d.Id(), err)
+	}
+
+	d.Set("name", resp.VoiceConnectorGroup.Name)
+
+	if err := d.Set("connector", flattenVoiceConnectorItems(resp.VoiceConnectorGroup.VoiceConnectorItems)); err != nil {
+		return diag.Errorf("error setting Chime Voice connector group items (%s): %s", d.Id(), err)
+	}
+	return nil
+}
+
+func resourceAwsChimeVoiceConnectorGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).chimeconn
+
+	input := &chime.UpdateVoiceConnectorGroupInput{
+		Name:                  aws.String(d.Get("name").(string)),
+		VoiceConnectorGroupId: aws.String(d.Id()),
+	}
+
+	if d.HasChange("connector") {
+		if v, ok := d.GetOk("connector"); ok {
+			input.VoiceConnectorItems = expandVoiceConnectorItems(v.([]interface{}))
+		}
+	} else if !d.IsNewResource() {
+		input.VoiceConnectorItems = make([]*chime.VoiceConnectorItem, 0)
+	}
+
+	if _, err := conn.UpdateVoiceConnectorGroupWithContext(ctx, input); err != nil {
+		if isAWSErr(err, chime.ErrCodeNotFoundException, "") {
+			log.Printf("[WARN] Chime Voice conector group %s not found", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("error updating Voice connector group (%s): %s", d.Id(), err)
+	}
+
+	return resourceAwsChimeVoiceConnectorGroupRead(ctx, d, meta)
+}
+
+func resourceAwsChimeVoiceConnectorGroupDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).chimeconn
+
+	if v, ok := d.GetOk("connector"); ok && len(v.([]interface{})) > 0 {
```

Comment:
Why does the update need to happen before the deletion?

Reply:
We must remove added connectors from the group before we can delete it.
https://docs.aws.amazon.com/chime/latest/APIReference/API_DeleteVoiceConnectorGroup.html

Reply:
That makes sense! I misunderstood what was being done there.